src/www/build.action.mjs

```diff
@@ -0,0 +1,76 @@
+// Copyright 2022 The Outline Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import fs from 'fs/promises';
+import webpack from 'webpack';
+import url from 'url';
+import path from 'path';
+
+import electronConfig from './webpack_electron.mjs';
+import cordovaConfig from './webpack_cordova.mjs';
+
+import {getBuildParameters} from '../../scripts/get_build_parameters.mjs';
+import {getBuildEnvironment} from '../../scripts/get_build_environment.mjs';
+import {getWebpackBuildMode} from '../../scripts/get_webpack_build_mode.mjs';
+
+const webpackPromise = webpackConfig =>
```

Comment:
In general it's better to avoid running code in the global namespace outside a function scope. It's hard to track.

Can you move this to inside the main?

Reply:
Done. FWIW I think the way I had it is slightly more performant, but negligible.